src/rust/aggregator/src/pipelines/spreads.rs

```diff
@@ -0,0 +1,373 @@
+                    if max < &element.price {
+                        max_bids.insert(element.market_id, element.price);
+                    }
+                } else {
+                    max_bids.insert(element.market_id, element.price);
+                }
+            }
+        }
+
+        (min_asks, max_bids)
+    }
+}
+#[async_trait::async_trait]
+impl Pipeline for Spreads {
+    fn model_name(&self) -> String {
+        String::from("Spreads")
+    }
+
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::from_std(TIMEOUT).unwrap()
+                < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> PipelineAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(TIMEOUT)
+    }
+
+    async fn process_and_save_internal(&mut self) -> PipelineAggregationResult {
+        let mut transaction = create_repeatable_read_transaction(&self.pool).await?;
+        let last_indexed_timestamp =
+            sqlx::query_file!("sqlx_queries/spreads/get_last_indexed_timestamp.sql")
+                .fetch_optional(&mut transaction as &mut PgConnection)
+                .await
+                .map_err(to_pipeline_error)?;
+        if last_indexed_timestamp.is_none() {
+            sqlx::query_file!("sqlx_queries/spreads/init_last_indexed_timestamp.sql")
+                .execute(&mut transaction as &mut PgConnection)
+                .await
+                .map_err(to_pipeline_error)?;
+        }
+        let last_indexed_timestamp =
+            sqlx::query_file!("sqlx_queries/spreads/get_last_indexed_timestamp.sql")
+                .fetch_optional(&mut transaction as &mut PgConnection)
+                .await
+                .map_err(to_pipeline_error)?;
+        if last_indexed_timestamp.is_none() {
+            return Ok(());
+        }
+        let last_indexed_timestamp = last_indexed_timestamp.unwrap().time;
+        let latest_event_timestamp =
+            sqlx::query_file!("sqlx_queries/spreads/get_latest_event_timestamp.sql")
+                .fetch_one(&mut transaction as &mut PgConnection)
+                .await
+                .map_err(to_pipeline_error)?
+                .time;
+        let latest_event_timestamp = if let Some(e) = latest_event_timestamp {
+            e
+        } else {
+            return Ok(());
+        };
+        let intervals = (latest_event_timestamp - last_indexed_timestamp).num_minutes() as usize;
+        let mut timestamps = Vec::with_capacity(intervals);
+        for i in 0..intervals {
+            timestamps.push(last_indexed_timestamp + chrono::Duration::minutes(i as i64 + 1));
+        }
+        if timestamps.len() == 0 {
+            return Ok(());
+        }
+        let mut state = self.state.clone().unwrap_or_default();
```

Comment:
Is there anyway we can mutably borrow this instead of cloning? Pass-by-value here could get expensive